python/cudf/cudf/_lib/column.pyx

```diff
@@ -372,14 +373,16 @@ cdef class Column:
         self._children = None
         self._data = None
 
-        return mutable_column_view(
+        cdef mutable_column_view out = mutable_column_view(
             dtype,
             self.size,
             data,
             mask,
             c_null_count,
             offset,
             children)
+        self._xs.push_back(<weak_ptr[int]>(out._x))
+        return out
```

Comment:
This seems backwards. I would expect a single a `Column` to effectively contain a `shared_ptr<Column>` to itself and `column_view` would hold a copy of that that `shared_ptr<Column>` such that it will refcount the number of `column_view`s referencing that `Column`. The `Column::spillable` check would just check `shared_ptr::use_count() == 1`, indicating that there are no `column_view`s to this column.

Reply:
This would require keeping a reference to a `PyObject` in `cudf::column_view`. It's doable, but I think there are lots of pitfalls there.

Reply:
If the column_view has a `shared_ptr<void>` with a custom, type-erased deleter, it doesn't really matter what the referenced object is.